external/avro/src/test/scala/org/apache/spark/sql/avro/AvroSuite.scala

```diff
@@ -692,7 +692,7 @@ class AvroSuite extends QueryTest with SharedSQLContext with SQLTestUtils {
           |  "type" : "record",
           |  "name" : "test_schema",
           |  "fields" : [{
-          |    "name": "enum",
+          |    "name": "Suit",
```

Comment:
`enum` type name and the field name (of that `enum` type) can be different. We should not assume that this line is equal to the line 697.

Reply:
In the original test case, the avro schema is `enum: union`, while the catalyst schema is `StructType(Seq(StructField("Suit", StringType, true)))`. Thus the field names can't match with each other.
We should fix such case.

Reply:
There are two names in Avro enum type. The first one is the field name used in catalyst, and the second one is the enum type name which is never read in catalyst format. I made the test case clearer to distinguish those two cases.

As a result, I'll argue that in the original code, the test case has a bug, and the test passes just because we never check if the field names are matching.

Reply:
Ya. The newly updated test code looks more clear.